scrape/scrape.go

```diff
@@ -807,10 +809,20 @@ type targetScraper struct {
 
 var errBodySizeLimit = errors.New("body size limit exceeded")
 
-const (
-	scrapeAcceptHeader             = `application/openmetrics-text;version=1.0.0,application/openmetrics-text;version=0.0.1;q=0.75,text/plain;version=0.0.4;q=0.5,*/*;q=0.1`
-	scrapeAcceptHeaderWithProtobuf = `application/vnd.google.protobuf;proto=io.prometheus.client.MetricFamily;encoding=delimited,application/openmetrics-text;version=1.0.0;q=0.8,application/openmetrics-text;version=0.0.1;q=0.75,text/plain;version=0.0.4;q=0.5,*/*;q=0.1`
-)
+func acceptHeader(sps []config.ScrapeProtocol) string {
+	var vals []string
+	for _, sp := range sps {
+		switch {
+		case sp.Equals(config.PrometheusText):
+			vals = append(vals, "text/plain;version=0.0.4;q=0.5")
+		case sp.Equals(config.PrometheusProto):
+			vals = append(vals, "application/vnd.google.protobuf;proto=io.prometheus.client.MetricFamily;encoding=delimited")
+		case sp.Equals(config.OpenMetricsText):
+			vals = append(vals, "application/openmetrics-text;version=1.0.0,application/openmetrics-text;version=0.0.1;q=0.75,*/*;q=0.1")
```

Comment:
This is 2 scrape protocols.

We should split 1.0.0 and 0.1.0

Reply:
Unless you mean that */* 0.1 thingy, maybe that's something else (not sure what tbh)

Reply:
Well , I think scrape protocols selection is an advanced setting, and I would like to be able to select between openmetrics 1.0 or 0.1

Reply:
That means numerous enum items - we can do that - will add (:

Reply:
Also quality should be respected (that gives actual order) https://www.rfc-editor.org/rfc/rfc9110.html#name-accept

Reply:
I consider changing to header values for more flexibility and simplicity:

<img width="1227" alt="image" src="https://github.com/prometheus/prometheus/assets/6950331/67519d7c-2f82-4d31-a5ff-c16134997d08">

WDYT?

Reply:
I think we should allow the user to specify the version. There might be a feature in newer versions that a user wants to switch on or off.

There might also be broken targets that behave weirdly, although they create the desired exposition format if we only send them the right accept header.

I would actually just let the user set the accept header. This is a power user setting anyway, we don't need a lot of guard railing here.

@roidelapluie what bad things do you have in mind if we let the user set the accept header themself?

Reply:
And about the quality AKA weight (I would prefer the latter, the RFC is using both, for some reason): If we allowed the user to set the accept header, it would be up to them. If we don't want to let them set the weight parameter, we have to do it ourselves to model the priority order the user has given. 

So in summary, I think we either do the enum approach, including versions, where the allowed items are fully fledged accept header components _without_ the weight (like in the screenshot above), so we set the weight ourselves according to the order in the config.

Or we let the user set the accept header directly, just one string, no validation, everything is up to the user including weight.

I currently prefer the latter, but @roidelapluie might convince me otherwise.

Reply:
Thanks @beorn7 - I tend to agree. 

The arguments against free-form text: 

1. UX. I personally have to search for 5m to find the exact header value and it's easy to mess it up. But user would need to do it only once.. 🤔  WIth enums users would need to search for some unique enum values too (!).
2. With enums we would clearly show what works (supported) with Prometheus. If someone tries using future protocol version - it's clear it's not possible without forking code or PR etc - so no support bugs with something very custom there.

What do I miss @roidelapluie ?
 
I am fine with both, but I am not fine WITHOUT this feature (: We could do "ScrapeAcceptHeader" string for now too as a first step,

Reply:
![image](https://github.com/prometheus/prometheus/assets/6950331/c347b415-12ca-4a12-b61a-16f898033271)